### ISSUE TITLE
parsing and saving CartesianRange

### ISSUE THREAD

**Issue by jingpengwu** [JuliaIO/JSON.jl#176]
CartesianRange was not correctly serialized and read.

```
julia> config = JSON.parse("""
       {"globalOffset":[0,0,0],"compression":"none","eltype":"UInt8","cuboidSize":[516,516,64],"prefix":"cuboid_","cartesianRange":{"start":{"I":[1,1,1]},"stop":{"I":[0,0,0]}},"dir":"gs://jpwu/test.bigarray.img/"}""", dicttype=Dict{Symbol, Any})
Dict{Symbol,Any} with 7 entries:
  :globalOffset   => Any[0,0,0]
  :compression    => "none"
  :eltype         => "UInt8"
  :cuboidSize     => Any[516,516,64]
  :prefix         => "cuboid_"
  :cartesianRange => Dict{Symbol,Any}(Pair{Symbol,Any}(:start,Dict{Symbol,Any}(Pair{Symbol,Any}(:I,An…
  :dir            => "gs://jpwu/test.bigarray.img/"

julia> config[:cartesianRange]
Dict{Symbol,Any} with 2 entries:
  :start => Dict{Symbol,Any}(Pair{Symbol,Any}(:I,Any[1,1,1]))
  :stop  => Dict{Symbol,Any}(Pair{Symbol,Any}(:I,Any[0,0,0]))

julia> config[:cartesianRange][:start]
Dict{Symbol,Any} with 1 entry:
  :I => Any[1,1,1]

```

**Comment by TotalVerb** [comment]
How should it be serialized?

**Comment by jingpengwu** [comment]
Should be serialized to be the same with `string(x)` transformation, then read using `eval(parse(x))` to get type of `CartesianRange`.


I have found a dirty solution. before serializing directly, I converted the CartesianRange to string, and then serialized.

```
cr = CartesianRange((1,1,1))
d[:cartesianRange] = string(cr)
```

after reading, use `eval(parse(d[:cartesianRange]))` to get type of `CartesianRange`

**Comment by TotalVerb** [comment]
Hi Jingpeng,JSON is not a typed serialization format, so I don't think your proposed method of serializing as a string is appropriate in general. On another note, how could one distinguish between a plain string and a CartesianRange?                                                                                                                                                                                                                From: Jingpeng WuSent: Wednesday, October 19, 2016 08:48To: JuliaIO/JSON.jlReply To: JuliaIO/JSON.jlCc: Fengyang Wang; CommentSubject: Re: [JuliaIO/JSON.jl] parsing and saving CartesianRange (#176)Should be serialized to be the same with string(x) transformation, then read using eval(parse(x)) to get type of CartesianRange.

I have found a dirty solution. before serializing directly, I converted the CartesianRange to string, and then serialized.

cr = CartesianRange((1,1,1))
d[:cartesianRange] = string(cr)


after reading, use eval(parse(d[:cartesianRange])) to get type of CartesianRange

—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/JuliaIO/JSON.jl","title":"JuliaIO/JSON.jl","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/JuliaIO/JSON.jl"}},"updates":{"snippets":[{"icon":"PERSON","message":"@jingpengwu in #176: Should be serialized to be the same with `string(x)` transformation, then read using `eval(parse(x))` to get type of `CartesianRange`.\r\n\r\n\r\nI have found a dirty solution. before serializing directly, I converted the CartesianRange to string, and then serialized.\r\n\r\n```\r\ncr = CartesianRange((1,1,1))\r\nd[:cartesianRange] = string(cr)\r\n```\r\n\r\nafter reading, use `eval(parse(d[:cartesianRange]))` to get type of `CartesianRange`\r\n\r\n"}],"action":{"name":"View Issue","url":"https://github.com/JuliaIO/JSON.jl/issues/176#issuecomment-254802315"}}}
